sympy/solvers/ode/single.py

```diff
@@ -1889,5 +1889,92 @@ def _get_general_solution(self, *, simplify_flag: bool = True):
         return fsol
 
 
+class Hypergeometric2nd(SingleODESolver):
```

Comment:
Most of the code in this class should be in functions in hypergeometric.py. The class should be completely minimal.